radar-upload-backend/src/main/java/org/radarbase/upload/dto/RecordMapperImpl.kt

```diff
@@ -46,7 +44,27 @@ class RecordMapperImpl: RecordMapper {
             modifiedDate = metadata.modifiedDate,
             committedDate = metadata.committedDate,
             logs = metadata.logs?.let {
-                LogsDto(url = "${uri.baseUri}/records/${metadata.id}/logs")
+                LogsDto(url = "${uri.baseUri}/records/${metadata.record.id}/logs")
             }
     )
+
+    override fun toMetadata(metadata: RecordMetadataDTO, origin: RecordMetadata) = RecordMetadata()
+            .also {
+                it.revision = metadata.revision
+                it.message = metadata.message
+                it.status = RecordStatus.valueOf(metadata.status)
+                if(origin != null) {
+                    it.createdDate = origin.createdDate
+                    it.committedDate = origin.committedDate
+                    it.modifiedDate = origin.modifiedDate
+                    it.logs = origin.logs
+                }
+            }
+
+    override fun fromMetadataToLogs(metadata: RecordMetadata) : LogsDto {
+        // read the first 5000 char from Clob
+        return LogsDto(
+                contents = metadata.logs?.logs?.getSubString(0, 5000),
```

Comment:
@blootsvoets I am not sure how do we want to provide logs content? should we read all of them from a Reader? or just first X number of characters etc or some other ideas?

Reply:
My intention was to store the complete logs. In any case, I do not expect such large logs to be stored, but having the full logs on errors is very useful.

Reply:
I not query the `logs` field at all in this call.